commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/threed/Plane.java

```diff
@@ -16,174 +16,193 @@
+    /**
+     * Build a plane from a point and a normal.
+     * 
+     * @param p         point belonging to the plane
+     * @param normal    normal direction to the plane
+     * @param precision precision context used to compare floating point values
+     * @return a new plane
+     * @throws IllegalNormException if the norm of the given values is zero, NaN, or infinite.
      */
-    public void reset(final Vector3D p, final Vector3D normal) {
-        setNormal(normal);
-        originOffset = -p.dot(w);
-        setFrame();
+    public static Plane fromPointAndNormal(final Vector3D p, final Vector3D normal, final DoublePrecisionContext precision) {
+        Vector3D w = normal.normalize();
+        double originOffset = -p.dot(w);
+        Vector3D projectedOrigin = calculateOrigin(w, originOffset);
+        Vector3D u = w.orthogonal();
+        Vector3D v = w.cross(u);
+        return new Plane(u, v, w, projectedOrigin, originOffset, precision);
     }
-
-    /** Reset the instance from another one.
-     * <p>The updated instance is completely independant of the original
-     * one. A deep reset is used none of the underlying object is
-     * shared.</p>
-     * @param original plane to reset from
+    
+    /**
+     * Build a plane from three points.
+     * <p>
+     * The plane is oriented in the direction of {@code (p2-p1) ^ (p3-p1)}
+     * </p>
+     * 
+     * @param p1        first point belonging to the plane
+     * @param p2        second point belonging to the plane
+     * @param p3        third point belonging to the plane
+     * @param precision precision context used to compare floating point values
+     * @return a new plane
+     * @throws IllegalNormException if the points do not constitute a plane
      */
-    public void reset(final Plane original) {
-        originOffset = original.originOffset;
-        origin       = original.origin;
-        u            = original.u;
-        v            = original.v;
-        w            = original.w;
+    public static Plane fromPoints(final Vector3D p1, final Vector3D p2, final Vector3D p3,
+            final DoublePrecisionContext precision) {
+        return Plane.fromPointAndNormal(p1, p2.subtract(p1).cross(p3.subtract(p1)), precision);
     }
 
-    /** Set the normal vactor.
-     * @param normal normal direction to the plane (will be copied)
-     * @exception IllegalNormException if the normal norm zero, NaN, or infinite
+    /**
+     * Copy plane.
+     * <p>
+     * The instance created is completely independent of the original one. A deep
+     * copy is used, none of the underlying object are shared.
+     * </p>
+     * 
+     * @param plane plane to copy
+     * @return a new plane
      */
-    private void setNormal(final Vector3D normal) {
-        final double norm = Vectors.checkedNorm(normal);
-
-        w = Vector3D.linearCombination(1.0 / norm, normal);
+    public static Plane of(final Plane plane) {
```

Comment:
done

Reply:
We may have miscommunicated on this. I was picturing that we don't need the `Plane.of(Plane)` method at all. Since `Plane` instances are immutable there's really no point to making a copy of one. I would keep the `final` modifier on parameters throughout, though.